File: lista-de-exercicios/exercicio-13/Aluno.ts
export class Aluno {
    private nota: number;
    private notas: Array<number>;
    private media: number;
    private status: string;

    public constructor (nota: number) {
        this.notas = [this.setNota(nota)];
        this.getMedia();
        this.getStatus();
    }

    public setNota(nota: number): number {
        this.nota = nota;
        if (nota <= 0)
            nota = 0;
        else if (nota >= 10)
            nota = 10;
        return nota;
    }

    public adicionarNota(nota: number) {
        this.notas.push(nota);
    }

    public getNotas() {
        return this.notas;
    }

    public getMedia() {
        let soma = 0;
        for (let i = 0; i < this.notas.length; i ++) {
            soma += this.notas[i];
        }
        this.media = soma / this.notas.length;
        return this.media;
    }

    public getStatus() {
        this.getMedia();
        if (this.media > 6)
            this.status = 'Aprovado';
        else if (this.media >= 4)
            this.status = 'Verificacao Suplementar';
        else this.status = 'Reprovado';
        return this.status;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio13/Aluno.java
package org.example.exercicio13;

import java.util.ArrayList;

public class Aluno {
    private double nota;
    private ArrayList<Double> notas;
    private double media;
    private String status;

    public Aluno() {
        this.notas = new ArrayList<>();
    }

    public void adicionarNota(double nota) {
        if (nota < 0 || nota > 10) {
            throw new IllegalArgumentException("A nota deve estar entre 0 e 10.");
        }
        this.notas.add(nota);
    }

    public ArrayList<Double> getNotas() {
        return notas;
    }

    public double calcularMedia() {
        double soma = 0;
        for (double nota : notas)
            soma += nota;
        this.media = soma / this.notas.size();
        return this.media;
    }


    public String calcularStatus() {
        if (this.media > 6)
            this.status = "Aprovado";
        else if (this.media >= 4)
            this.status = "Verificação Suplementar";
        else this.status = "Reprovado";
        return this.status;
    }

    public double getMedia() {
        return this.media;
    }

    public String getStatus() {
        return this.status;
    }
}
